src/20240411_Multi_SecurityBudgetRequestDec23AndRobotRefill/AaveV3Ethereum_SecurityBudgetRequestDec23AndRobotRefill_20240411.t.sol

```diff
@@ -0,0 +1,151 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {AaveV3Ethereum} from 'aave-address-book/AaveV3Ethereum.sol';
+import {AaveV2EthereumAssets} from 'aave-address-book/AaveV2Ethereum.sol';
+import {GovernanceV3Ethereum} from 'aave-address-book/GovernanceV3Ethereum.sol';
+
+import {ProtocolV3TestBase} from 'aave-helpers/ProtocolV3TestBase.sol';
+import {AaveV3Ethereum_SecurityBudgetRequestDec23AndRobotRefill_20240411} from './AaveV3Ethereum_SecurityBudgetRequestDec23AndRobotRefill_20240411.sol';
+import {IERC20} from 'solidity-utils/contracts/oz-common/interfaces/IERC20.sol';
+
+/**
+ * @dev Test for AaveV3Ethereum_SecurityBudgetRequestDec23AndRobotRefill_20240411
+ * command: make test-contract filter=AaveV3Ethereum_SecurityBudgetRequestDec23AndRobotRefill_20240411
+ */
+contract AaveV3Ethereum_SecurityBudgetRequestDec23AndRobotRefill_20240411_Test is
+  ProtocolV3TestBase
+{
+  AaveV3Ethereum_SecurityBudgetRequestDec23AndRobotRefill_20240411 internal proposal;
+
+  event KeeperRefilled(uint256 indexed id, address indexed from, uint96 indexed amount);
+
+  function setUp() public {
+    vm.createSelectFork(vm.rpcUrl('mainnet'), 19631594);
+    proposal = new AaveV3Ethereum_SecurityBudgetRequestDec23AndRobotRefill_20240411();
+  }
+
+  /**
+   * @dev executes the generic test suite including e2e and config snapshots
+   */
+  function test_defaultProposalExecution() public {
+    defaultTest(
+      'AaveV3Ethereum_SecurityBudgetRequestDec23AndRobotRefill_20240411',
+      AaveV3Ethereum.POOL,
+      address(proposal)
+    );
+  }
+
+  function test_consistentBalances() public {
+    uint256 collectorAUsdcBalanceBefore = IERC20(AaveV2EthereumAssets.USDC_A_TOKEN).balanceOf(
+      address(AaveV3Ethereum.COLLECTOR)
+    );
+    uint256 collectorAUsdtBalanceBefore = IERC20(AaveV2EthereumAssets.USDT_A_TOKEN).balanceOf(
+      address(AaveV3Ethereum.COLLECTOR)
+    );
+    uint256 collectorALinkBalanceBefore = IERC20(AaveV2EthereumAssets.LINK_A_TOKEN).balanceOf(
+      address(AaveV3Ethereum.COLLECTOR)
+    );
+
+    // Validate the Collector has enough aUSDC v2
+    assertGe(collectorAUsdcBalanceBefore, proposal.USDC_AMOUNT_REIMBURSEMENT());
```

Comment:
not sure if its needed as the test will fail on execution, but definitely up to you